concrete/src/File/ImportProcessor/SvgSanitizerProcessor.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Concrete\Core\File\ImportProcessor;
+
+use Concrete\Core\Entity\File\Version;
+use Concrete\Core\File\Image\Svg\Sanitizer;
+use Concrete\Core\File\Image\Svg\SanitizerOptions;
+use Concrete\Core\File\Type\Type;
+
+class SvgSanitizerProcessor implements ProcessorInterface
+{
+    /**
+     * SVG sanitizer.
+     *
+     * @var \Concrete\Core\File\Image\Svg\Sanitizer
+     */
+    protected $sanitizer;
+
+    /**
+     * SVG sanitizer options.
+     *
+     * @var \Concrete\Core\File\Image\Svg\SanitizerOptions|null
+     */
+    private $sanitizerOptions;
+
+    /**
+     * Initialize the instance.
+     *
+     * @param \Concrete\Core\File\Image\Svg\Sanitizer $sanitizer
+     */
+    public function __construct(Sanitizer $sanitizer)
+    {
+        $this->sanitizer = $sanitizer;
+    }
+
+    /**
+     * Set the sanitizer options.
+     *
+     * @param \Concrete\Core\File\Image\Svg\SanitizerOptions $options
+     *
+     * @return $this
+     */
+    public function setSanitizerOptions(SanitizerOptions $options)
+    {
+        $this->sanitizerOptions = $options;
+
+        return $this;
+    }
+
+    /**
+     * Get the sanitizer options.
+     *
+     * @return \Concrete\Core\File\Image\Svg\SanitizerOptions
+     */
+    public function getSanitizerOptions()
+    {
+        if ($this->sanitizerOptions === null) {
+            $this->sanitizerOptions = SanitizerOptions::create();
+        }
+
+        return $this->sanitizerOptions;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Concrete\Core\File\ImportProcessor\ProcessorInterface::shouldProcess()
+     */
+    public function shouldProcess(Version $version)
+    {
+        $versionTypeObject = $version->getTypeObject();
+
+        return $versionTypeObject->getGenericType() == Type::T_IMAGE && $versionTypeObject->isSVG();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Concrete\Core\File\ImportProcessor\ProcessorInterface::process()
+     */
+    public function process(Version $version)
+    {
+        $resource = $version->getFileResource();
+        $originalData = $resource->read();
```

Comment:
Does this handle SVGs that are larger than 1/2 the maximum memory? It might be better to do this in a stream rather than by reading the entire file into memory

Reply:
Well, the system already loads images into memory in previous steps of the import function. If we get here, we should have enough memory.

Reply:
Fair enough. It is totally possible to stream process these files with the dependencies we have for what it's worth.